pkg/ddevapp/compose_yaml.go

```diff
@@ -78,17 +87,54 @@ func (app *DdevApp) WriteDockerComposeYAML() error {
 		return err
 	}
 
+	// Run docker-compose config on the full rendered yaml file again after fixes.
+	// This action will override the existing file with the new one.
+	fullContentsFinal, _, err := dockerutil.ComposeCmd(&dockerutil.ComposeCmdOpts{
+		ComposeFiles: []string{app.DockerComposeFullRenderedYAMLPath()},
+		Action:       append(action, "config"),
+	})
```

Comment:
I have to run `docker-compose config` again because I add `env_file` to services and I need to convert it to an `environment` stanza.

This is the second thing I don't like about this PR.

Reply:
I have an idea how to get rid of this second `config` call.

I should manually adjust `environment` stanza, instead of using `env_file` stanza, and it will also fix the problem with:

```
!!!warning "Environment variables cannot be overridden in some cases"
```